src/TConsole.cpp

```diff
@@ -772,14 +777,31 @@ void TConsole::closeEvent( QCloseEvent *event )
 
     if( profile_name != "default_host" && ! mUserAgreedToCloseConsole )
     {
-        ASK: int choice = QMessageBox::question( this, "Exiting Session: Question", "Do you want to save the profile "+profile_name, QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel );
+        QCheckBox *cb = new QCheckBox( tr("Save the Window Layout before closing this profile.") );
```

Comment:
I boxed with this for some time.  I would have liked to have a layout for each profile but that complicated things and caused layout management issues in multi-play, like you suspect.  
  
The scope is application-wide, and we couldn't really change that without writing some custom parser to replace the `saveState()` and `restoreState()` methods.  Per the documentation I've read, it includes all window objects that are handled by `QMainWindow::saveState()` and `QMainWindow::restoreState()` and as far as I know the requirement is that the widget/window objects need to have an `objectName` set and that name must be the same between saves and loads or the data isn't used.